identity_iota_core/packages/iota_identity/sources/proposals/config.move

```diff
@@ -21,59 +21,38 @@ public fun propose_modify<V>(
     multi: &mut Multicontroller<V>,
     cap: &DelegationToken,
     expiration: Option<u64>,
-    mut threshold: Option<u64>,
+    threshold: Option<u64>,
     controllers_to_add: VecMap<address, u64>,
     controllers_to_remove: vector<ID>,
     controllers_to_update: VecMap<ID, u64>,
     ctx: &mut TxContext,
 ): ID {
     let mut max_votes = 0;
-    let (mut cs, mut vps) = controllers_to_update.into_keys_values();
-    while (!cs.is_empty()) {
-        let c = cs.pop_back();
-        let vp = vps.pop_back();
+    let (cs, vps) = controllers_to_update.into_keys_values();
+    vector::zip_do!(cs, vps, |c, vp| {
         assert!(multi.controllers().contains(&c), ENotMember);
         max_votes = max_votes + vp;
-    };
-    let (_, mut voting_powers) = controllers_to_add.into_keys_values();
-    let mut voting_power_increase = 0;
-    while (!voting_powers.is_empty()) {
-        let voting_power = voting_powers.pop_back();
-
-        voting_power_increase = voting_power_increase + voting_power;
-    };
-    voting_powers.destroy_empty();
+    });
```

Comment:
Most probably missing something, but was wondering why this part isn't using `fold!` to accumulate the max_votes, similar to how the next two sections do.